current/runtime/fees.md

```diff
@@ -4,118 +4,151 @@ lang: en
     MaxOperational,
-    /// An operational dispatch with no weight.
-    FreeOperational,
+    FixedMandatory(Weight),
 }
 ```
 
-This enum groups all dispatches into _normal_ or _operational_ and gives them a fixed weight. This
-means that the arguments of a dispatchable function do not affect the weight. A dispatch classified
-as _operational_ is exempt from paying both the `base_fee` and the `length_fee`.
-
-### Free Normal
-
-This means that this function has no weight. It will not contribute to block fullness at all, and no
-weight-fee is applied.
+In order to delineate their purposes, the enums in this group are separated into _dispatch classes_,
+which are also defined by an enum:
 
 ```rust
-#[weight = SimpleDispatchInfo::FreeNormal]
-pub fn some_normal_function_light() { noop(); }
+pub enum DispatchClass {
+    Normal,
+    Operational,
+    Mandatory,
+}
 ```
 
-### Fixed Normal
+### Normal Dispatches
 
-This function will always have a weight `10`.
+Dispatches in this class represent normal user-triggered transactions. These types of dispatches may
+only consume a portion of a block's total weight limit; this portion can be found by examining the
+[`AvailableBlockRatio`](https://substrate.dev/rustdocs/master/frame_system/trait.Trait.html#associatedtype.AvailableBlockRatio).
+Normal dispatches are assigned a priority that is proportional to their weight.
+
+#### `FixedNormal`
+
+Describes a normal function that will always have the specified weight.
 
 ```rust
 #[weight = SimpleDispatchInfo::FixedNormal(10)]
 pub fn some_normal_function_heavy() { some_computation(); }
 ```
 
-### Fixed Operational
+#### `MaxNormal`
+
+This is equivalent to `SimpleDispatchInfo::FixedNormal(Weight::max_value())`.
 
-This function will have a fixed weight but can consume the reserved operational portion as well.
+#### `InsecureFreeNormal`
+
+This means that the function has no weight; it will not contribute to block fullness at all and no
+weight-fee is applied. Although the `base_fee` and `length_fee` still need to be paid, as the name
+indicates a lack of weight fees also implies a lack of security, so dispatches of this type should
+be used with care.
 
 ```rust
-#[weight = SimpleDispatchInfo::FixedOperational(20)]
-pub fn mission_critical_function() { some_sudo_op(); }
+#[weight = SimpleDispatchInfo::InsecureFreeNormal]
+pub fn some_normal_function_light() { noop(); }
 ```
 
-### Default
+### Operational Dispatches
+
+Operational dispatches are those that go beyond user-triggered transactions; they are dispatches
+that are necessary to provide network capabilities. These types of dispatches are not bound by the
+[`AvailableBlockRatio`](https://substrate.dev/rustdocs/master/frame_system/trait.Trait.html#associatedtype.AvailableBlockRatio),
+which means they may consume a block's entire weight limit. Dispatches in this class are given maximum
+priority and are exempt from paying the `base_fee` and `length_fee`.
```

Comment:
I would be more specific than just "user-triggered". Is a fisherman a user? Sort of. Also, Substrate is generic, we may not have any regular-user-dispatchable `Operational`s in Polkadot, but someone developing their own runtime and/or pallet might have a valid reason to make some tx `Operational`.

The part about `AvailableBlockRatio` could be reworded. Without actually reading that rustdoc link, it sounds like these `Operationals` aren't bound by much of a limit at all.